### PR TITLE
Team33 37 signup backend

### PR DESCRIPTION
**Pull request by ValeSer**
Sign up with just email and password

### CODE REVIEW

File: backend/src/models/farmer.ts
Comment (TianshengC) [inline backend/src/models/farmer.ts:18]:
```suggestion
import { Schema, model, Model, Document } from 'mongoose';

// 1. Create an interface representing a document in MongoDB.
interface IFarmer extends Document {
  email: string;
  password: string;
}

// 2. Create a Schema corresponding to the document interface.
const farmerSchema = new Schema<IFarmer>({
  email: { type: String, required: true, unique: true  },
  password: { type: String, required: true }
})

// 3. Create a Model.
const Farmer: Model<IFarmer> = model<IFarmer>('farmers', farmerSchema);

export { Farmer };
```
1. connect is not used so deleted.
2. I add Document and Model type for clarification.
3. I add unique for the email attribute.

File: backend/src/index.ts
Comment (TianshengC) [inline backend/src/index.ts:22]: const PORT = process.env.BACK_PORT || 8081;
app.listen(PORT, () => {
  console.log(`server running : http://localhost:${PORT}`);
});

Please use the env variable.

File: backend/src/models/farmer.ts
Comment (TianshengC) [reply backend/src/models/farmer.ts:18]: Other suggestions about the **mongoConfig.ts:**

import { Schema } from 'mongoose'; delete this line(2) because Schema is not used.

const mongoDbUrl = 'mongodb://0.0.0.0/vegilicious';   
const mongoDbUrl = process.env.MONGODB_URL
Please use .env variable.

**Review by TianshengC — CHANGES_REQUESTED** [review]
Thanks, Valeria, the code functions well and passes the test. Good job!

Meanwhile, we can still improve the quality of the code. Please check my comments.

File: backend/src/index.ts
Comment (ValeSer) [reply backend/src/index.ts:10]: It is necessary to make the connection happen. When I removed it, db did not get read. It is a work around, we need to refactor connection in the future

File: backend/src/controllers/signupController.ts
Comment (ValeSer) [reply backend/src/controllers/signupController.ts:36]: I like the refactoring with handling error, apart from the error message returned, I would avoid to expose info about our database to the client. Just res.status(500) would do the job :)

File: backend/src/index.ts
Comment (ValeSer) [reply backend/src/index.ts]: Done :)

**Comment by ValeSer** [comment]
Changes have been made, waiting for approval after new review, thank you :)

**Review by TianshengC — APPROVED** [review]

**Pull request merged**
